src/main/java/com/microsoft/lst_bench/Driver.java

```diff
@@ -142,6 +145,9 @@ public static void main(String[] args) throws Exception {
     final BenchmarkRunnable experiment =
         new LSTBenchmarkExecutor(idToConnectionManager, benchmarkConfig, telemetryRegistry);
     experiment.execute();
+
+    // Remove hook.
+    Runtime.getRuntime().removeShutdownHook(telemetryHook);
```

Comment:
I think rather than adding and removing the hook, we can modify the `flush` method to clear the buffer / list of events. I think this aligns better with the expected behavior of such a method.

Reply:
I changed the flush method to reset the buffer after writing and added a finally clause to post-phase processing to flush telemetry after each phase. I still think an additional hook is useful since someone might cancel execution mid-phase in which case we would not record telemetry otherwise.

Btw, this also solves a bug. I think if you have multiple repetitions right now, it would have logged the same telemetry multiple times since the event list was not cleared between repetitions.

Reply:
Thanks @anjagruenheid . Yes, I think this will solve that potential issue. I think removing the hook at the end of execution is not necessary anymore though (since you are not going to flush the same telemetry twice)?

Reply:
Okay, removed the removal. :)